### PR TITLE
Fix `xml:filter` return type on non `xml:Element` singleton values

### LINKED ISSUE #42817: [Bug]: Equality checks on `xml:filter` function results on single `xml:PI`, `xml:Text` and `xml:Comment` doesn't work as expected
### Description

Consider the following code, 

```ballerina
import ballerina/io;

public function main() {
    // this works as expected
    xml<xml:Element> x1 = xml `<a/>`;
    xml r = xml:filter(x1, v => true); 
    boolean b = r == xml `<a/>`;
    io:println(b); // true
    io:println(r); // <a/>


    // these doesn't work as expected
    xml<xml:ProcessingInstruction> x2 = xml `<?data?>`;
    r = xml:filter(x2, v => true); 
    b = r == xml `<?data?>`;
    io:println(b); // false
    io:println(r); // <?data ?>

    xml<xml:Comment> x3 = xml `<!--comment-->`;
    r = xml:filter(x3, v => true); 
    b = r == xml `<!--comment-->`;
    io:println(b); // false
    io:println(r); // <!--comment-->

    xml<xml:Text> x4 = xml `this is a text`;
    r = xml:filter(x4, v => true); 
    b = r == xml `this is a text`;
    io:println(b); // false
    io:println(r); // this is a text
}
```
The equality checks are expected to return `true` here.

### Steps to Reproduce

_No response_

### Affected Version(s)

2201.9.0

### OS, DB, other environment details and versions

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by poorna2152**
## Purpose
Fixes #42817

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Filter.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public static BXml filter(BXml x, BFunctionPointer<Object, Boolean> func) {
             func.asyncCall(args,
                       result -> {
                           if ((Boolean) result) {
-                              return ValueCreator.createXmlSequence(x);
+                              return x;
```
Comment (chiranSachintha) [inline langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Filter.java:60]: Is this correct? My concern is that the specification mentions that the `filter` function returns a new XML sequence containing items in parameter `x` for which function `func` evaluates to true. So, the value returned from the `filter` function should be an XML sequence. If that is the case, we need to fix this in the `isEqual` function in the type checker, where we check the tag of the type.

Comment (poorna2152) [reply langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Filter.java:60]: Under the [sequence value section](https://ballerina.io/spec/lang/master/#section_5.3) in the specification it is defined that, 
> Note that for a sequence consisting of a single item v is the same thing as v. A single item is a sequence.

Thus IMO it is correct here not to create a sequence.

Comment (chiranSachintha) [reply langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Filter.java:60]: @MaryamZi What is your opinion about this? 
`Note that for a sequence consisting of a single item, v is the same thing as v. A single item is a sequence.`
With this also, I feel that a single item should be treated as a sequence.

Comment (MaryamZi) [reply langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Filter.java:60]: From a user PoV, there's no difference. You can't differentiate between a sequence of length 1 and a singleton in Ballerina code. The implementation could choose either way as long as sequences of length 1 are handled correctly everywhere. I would probably go with not creating a sequence.

Comment (chiranSachintha) [reply langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Filter.java:60]: Got it. Thanks for the clarification, @poorna2152 So, we can keep the logic as it is now.

Comment (poorna2152) [reply langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Filter.java:60]: Ack

**Review by chiranSachintha — APPROVED** [review]

**Pull request merged**